node/node.cpp

```diff
@@ -200,7 +206,13 @@ node::find_session(const boost::asio::ip::tcp::endpoint& ep)
                 , this->options
                 , std::nullopt
                 , this->client_ctx);
-        session->open(this->websocket);
+        session->open(this->websocket, [self = shared_from_this(), ep](const boost::system::error_code& ec)
+        {
+            if (ec && self->error_callback)
+            {
+                self->error_callback(ep, ec);
```

Comment:
It seems that we only invoke this callback on failures that happen related to the handshake - given that it's being sued to trigger a view change, should we not also call it when the socket fails later, or even when it gracefully closes?

Reply:
My assumption was that we would try to reconnect and notice the problem there. I could add a socket error case too. In terms of gracefully closing couldn't a session time out and cause that legitimately?